ts/store/actions/deferred.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Action types and action creators related to deferred actions.
+ * A deferred action is an action dispatched before the user login
+ * but actually has to be dispatched once the has been logged in.
+ * (e.g. navigation coming from push notification or from deep link happened
+ * while the app is not in foreground and/or the user is not logged in)
```

Comment:
I'm not sure why we need such a generic mechanism for deferring multiple arbitrary actions, aren't we talking just about a single navigation action? what are the use cases for such a generic mechanism?

Reply:
The use cases, at this time, are the interaction with a push notification (message notification -> navigate to the message, maybe there will be others?) and navigation from a deep link.